alf/utils/common.py

```diff
@@ -1082,6 +1082,10 @@ def generate_alf_root_snapshot(alf_root, dest_path):
         dest_path (str): the path to generate a snapshot of ALF repo
     """
 
+    def _is_subdir(path, directory):
+        relative = os.path.relpath(path, directory)
+        return not relative.startswith(os.pardir)
```

Comment:
This does not always work. For example:

```
>>> os.path.relpath("/abc/def", "/abc")
'def'
```

Reply:
> This does not always work. For example:
> 
> ```
> >>> os.path.relpath("/abc/def", "/abc")
> 'def'
> ```

'def' is the expected result in this case. This function decides if 'path' is a subdirectory of 'directory'. If not, relative should always start with '..'

Reply:
You right. I was confused.